tests/util/checker/checker.go

```diff
@@ -52,22 +54,28 @@ func Check(paths []string, factory RulesFactory, whitelist *Whitelist, report *R
 }
 
 // fileCheck checks a file using the given rules, and write to the given Report.
-func fileCheck(path string, rules []Rule, whitelist *Whitelist, report *Report) {
+func fileCheck(path string, rules []Rule, report *Report) {
 	fs := token.NewFileSet()
-	astFile, err := parser.ParseFile(fs, path, nil, parser.Mode(0))
+	astFile, err := parser.ParseFile(fs, path, nil, parser.ParseComments)
 	if err != nil {
 		report.AddString(fmt.Sprintf("%v", err))
 		return
 	}
 	v := FileVisitor{
-		path:      path,
-		rules:     rules,
-		whitelist: whitelist,
-		fileset:   fs,
-		report:    report,
+		path:    path,
+		rules:   rules,
+		fileset: fs,
+		report:  report,
 	}
 	// Walk through the files
-	ast.Walk(&v, astFile)
+	// ast.Walk(&v, astFile)
+	for _, d := range astFile.Decls {
+		if fn, isFn := d.(*ast.FuncDecl); isFn && strings.HasPrefix(fn.Name.String(), "Test") {
```

Comment:
Theoretically the linter can apply to regular go files. (Perhaps we should think of a better name.)